x/concentrated-liquidity/fees.go

```diff
@@ -80,26 +80,18 @@ func (k Keeper) initializeFeeAccumulatorPosition(ctx sdk.Context, poolId uint64,
 
 	// assure that existing position has zero liquidity
 	if hasPosition {
-		positionSize, err := feeAccumulator.GetPositionSize(positionKey)
-		if err != nil {
-			return err
-		}
-
-		if !positionSize.IsZero() {
-			return fmt.Errorf("attempted to re-initialize fee accumulator position (%s) with non-zero liquidity", positionKey)
-		}
+		return fmt.Errorf("attempted to re-initialize fee accumulator position (%s) with non-zero liquidity", positionKey)
 	}
 
 	// initialize the owner's position with liquidity Delta and zero accumulator value
-	if err := feeAccumulator.NewPosition(positionKey, liquidityDelta, nil); err != nil {
+	if err := feeAccumulator.NewPosition(positionKey, sdk.ZeroDec(), nil); err != nil {
```

Comment:
Note: this is intentional. The initialization should happen to zero shares. Then, `updateFeeAccumulatorPosition` should add liquidity to an already initialized position